### PR TITLE
Use JSONC in web API (part 1)

### PR DESCRIPTION
**Pull request by underhood**
##### Summary
This implements some of the api calls using JSONC which allows us to modify, extend or encapsulate payloads any time we want + generating correct JSON every time + allows us to generate both minified and human readable forms depending on the need.

Currently API is implemented as is incl. its various quirks like setting strings "null" instead using JSON type NULL, dictionaries instead of arrays, objects containing only 1 member and other funny things.... this quirks/bugs are copied to maintain API the same

### CODE REVIEW

**Comment by underhood** [comment]
might change based on discussion here https://github.com/netdata/netdata/discussions/12365
compared to manual JSON build the JSON-C version replaces some analytics.XYZ with source data to have correct numbers even in case of analytics disabled -> This has to be fixed in non JSON-C version as well

File: web/api/formatters/rrdset2json.c [hunk 1]
```diff
@@ -163,3 +172,143 @@ void rrdset2json(RRDSET *st, BUFFER *wb, size_t *dimensions_count, size_t *memor
 
     rrdset_unlock(st);
 }
+#endif /* ENABLE_JSONC */
+
+
+#ifdef ENABLE_JSONC
+#define JSON_ADD_STRING(name, str, obj)                                                                                \
```
Comment (thiagoftsm) [inline web/api/formatters/rrdset2json.c:179]: We have an analogous definition in the file `web/api/formatters/charts2json.c`, please, let us centralize this an `JSON_ADD_INT` in a header.

File: web/api/formatters/rrdset2json.c [hunk 2]
```diff
@@ -163,3 +172,143 @@ void rrdset2json(RRDSET *st, BUFFER *wb, size_t *dimensions_count, size_t *memor
...
+        // which also creates dictionary with fixed one item
+        // funny but... (maybe /api/v2 in future :) )
+        json_object *tmp_obj = json_object_new_object();
+        JSON_ADD_STRING("name", rd->name, tmp_obj);
+        json_object_object_add(obj, rd->id, tmp_obj);
+    }
+    json_object_object_add(j, "dimensions", obj);
+
+    buffer_flush(buf);
+    health_api_v1_chart_custom_variables2json(st, buf); // TODO jsonc-ify this
```
Comment (thiagoftsm) [inline web/api/formatters/rrdset2json.c:274]: I remembered now a PR that you review, should we create an issue instead to keep a `TODO` here?

File: web/api/web_api_v1.c [hunk 3]
```diff
@@ -934,10 +941,285 @@ inline void host_labels2json(RRDHOST *host, BUFFER *wb, size_t indentation) {
     netdata_rwlock_unlock(&host->labels.labels_rwlock);
     rrdhost_unlock(host);
 }
+#endif /* ENABLE_JSONC */
 
 extern int aclk_connected;
-inline int web_client_api_request_v1_info_fill_buffer(RRDHOST *host, BUFFER *wb)
+#ifdef ENABLE_JSONC
+#define JSON_ADD_STRING(name, str, obj)                                                                                \
```
Comment (thiagoftsm) [inline web/api/web_api_v1.c:948]: As I already wrote above, we have the same definition here.

File: web/api/web_api_v1.c [hunk 4]
```diff
@@ -934,10 +941,285 @@ inline void host_labels2json(RRDHOST *host, BUFFER *wb, size_t indentation) {
...
+    if (aclk_connected) {
+        JSON_ADD_BOOL("aclk-available", 1, j)
+#ifdef ENABLE_NEW_CLOUD_PROTOCOL
+        JSON_ADD_STRING("aclk-available-protocol", aclk_use_new_cloud_arch ? "New" : "Legacy", j);
+#else
+        JSON_ADD_STRING("aclk-available-protocol", "Legacy", j);
+#endif
+    } else
+#endif
+        JSON_ADD_BOOL("aclk-available", 0, j)
```
Comment (thiagoftsm) [inline web/api/web_api_v1.c:1166]: Please, fix tabulation here.

Comment (underhood) [reply web/api/web_api_v1.c:1166]: i consider it to be correct due to else on the line 1164? do you prefer it one tab left?

Comment (thiagoftsm) [reply web/api/web_api_v1.c:1166]: I am sorry, you can ignore this, please, only centralize the other defines.

File: web/api/formatters/rrdset2json.c (hunk 1 above)
Comment (underhood) [reply web/api/formatters/rrdset2json.c]: moved to common file

File: web/api/web_api_v1.c (hunk 3 above)
Comment (underhood) [reply web/api/web_api_v1.c]: moved to common file

**Pull request closed without merging**